state-synchronizer/src/counters.rs

```diff
@@ -19,9 +19,33 @@ pub const CHUNK_REQUEST_MSG_LABEL: &str = "chunk_request";
 pub const CHUNK_RESPONSE_MSG_LABEL: &str = "chunk_response";
 
 // version type labels
-pub const COMMITTED_VERSION_LABEL: &str = "committed"; // Version of latest ledger info committed.
-pub const SYNCED_VERSION_LABEL: &str = "synced"; // Version of most recent txn that was synced (even if it is not backed by an LI)
-pub const TARGET_VERSION_LABEL: &str = "target"; // Version a node is trying to catch up to
+pub fn set_version(version_type: VersionType, version: u64) {
+    VERSION
+        .with_label_values(&[version_type.as_str()])
+        .set(version as i64)
+}
+
+pub enum VersionType {
+    /// Version of latest ledger info committed.
+    Committed,
+    /// Highest known version or version proceeding it
+    Highest,
```

Comment:
Nit: Perhaps rename "Highest" to "HighestKnown" to make it explicit? "Highest" by itself seems a little ambiguous to me.